recursio-client/Rendering/VisibilityLight.gd

```diff
@@ -1,8 +1,34 @@
 tool
 extends SpotLight
-
-
+var _currently_toggling = false
+var _toggle_goal = -1
+var _toggle_start = -1
+var _time_since_toggle_start = -1
+onready var _toggle_time = Constants.get_value("visibility", "light_toggle_time")
+onready var _spot_range = Constants.get_value("visibility", "spot_range")
 func _ready():
 	$SightLight.light_energy = light_energy
 	$SightLight.spot_angle = spot_angle
-	$SightLight.spot_range = Constants.get_value("visibility", "spot_range")
+	spot_range = 0
+	$SightLight.spot_range = spot_range
+
+func _process(delta):
```

Comment:
All of this could've probably been simplified with a Tween node, but this is fine too. However, a comment saying that this is just lerping between on and off would be helpful, since it took me some time to realize :sweat_smile:

Reply:
Sorry, will add a comment.